imap_processing/tests/mag/test_mag_l1b.py

```diff
@@ -94,10 +95,10 @@ def test_mag_compression_scale(mag_l1a_dataset):
     mag_l1a_dataset["vectors"][2, :] = np.array([1, 1, 1, 0])
     mag_l1a_dataset["vectors"][3, :] = np.array([1, 1, 1, 0])
 
-    mag_l1a_dataset["compression_flags"][0, :] = np.array([1, 16])
-    mag_l1a_dataset["compression_flags"][1, :] = np.array([0, 0])
-    mag_l1a_dataset["compression_flags"][2, :] = np.array([1, 18])
-    mag_l1a_dataset["compression_flags"][3, :] = np.array([1, 14])
+    mag_l1a_dataset["compression_flags"][0, :] = np.array([1, 16], dtype=np.int8)
```

Comment:
This change was to more closely imitate the actual data, which caused a bug with raising integers to negative powers.